ui/hooks/usePolling.test.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { renderHook, cleanup } from '@testing-library/react-hooks';
+import { Provider } from 'react-redux';
+import configureStore from '../store/store';
+import usePolling from './usePolling';
+
+describe('usePolling', () => {
+  // eslint-disable-next-line jest/no-done-callback
+  it('calls startPollingByNetworkClientId and callback option args with polling token when component instantiating the hook mounts', (done) => {
+    const mockStart = jest.fn().mockImplementation(() => {
+      return Promise.resolve('pollingToken');
+    });
+    const mockStop = jest.fn();
+    const networkClientId = 'mainnet';
+    const options = {};
+    const mockState = {
+      metamask: {},
+    };
+
+    const wrapper = ({ children }) => (
+      <>
+        <Provider store={configureStore(mockState)}>{children}</Provider>
+      </>
+    );
```

Comment:
We have test utility function `renderHookWithProvider` already defined. May be can be used here.

Reply:
Looking into this!

Reply:
Done here: https://github.com/MetaMask/metamask-extension/pull/23010/commits/02a0c73d28e7b61dbaeb91589d5bb9a9b98aa129